pages/settings.js

```diff
@@ -69,174 +100,92 @@ class SettingsForm extends React.Component {
-                      <th className="has-text-centered">Public</th>
-                      <th>Actions</th>
-                    </tr>
-                  </thead>
-                  <tbody>
-                    {isMembershipSet ? (
-                      userInfo.membership_set.map(item => (
-                        <tr key={item.id}>
-                          <td>{item.name}</td>
-                          <td>{item.title}</td>
-                          <td>{item.role_display}</td>
-                          <td className="has-text-centered">
-                            <Icon
-                              name={
-                                item.active
-                                  ? 'check-circle-green'
-                                  : 'x-circle-red'
-                              }
-                              alt={item.active ? 'active' : 'inactive'}
-                            />
-                          </td>
-                          <td className="has-text-centered">
-                            <Icon
-                              name={
-                                item.public
-                                  ? 'check-circle-green'
-                                  : 'x-circle-red'
-                              }
-                              alt={item.public ? 'public' : 'not public'}
-                            />
-                          </td>
-                          <td className="buttons">
-                            <Link
-                              route="club-view"
-                              params={{ club: String(item.code) }}
-                            >
-                              <a className="button is-small is-link">View</a>
-                            </Link>
-                            {item.role <= ROLE_OFFICER && (
-                              <Link
-                                route="club-edit"
-                                params={{ club: String(item.code) }}
-                              >
-                                <a className="button is-small is-success">
-                                  Edit
-                                </a>
-                              </Link>
-                            )}
-                          </td>
-                        </tr>
-                      ))
-                    ) : (
-                      <tr>
-                        <td className="has-text-grey" colSpan="4">
-                          You are not a member of any clubs yet.
-                        </td>
-                      </tr>
-                    )}
-                  </tbody>
-                </table>
-              </div>
-            ),
-          },
-        ],
-      },
-    ]
+  leaveClub(club) {
+    if (confirm(`Are you sure you want to leave ${club.name}? You cannot add yourself back into the club.`)) {
+      doApiRequest(`/clubs/${club.code}/members/${this.props.userInfo.username}`, {
```

Comment:
Same thing here—would be good to parse out the `username` before putting it in the string to cut down on length + improve readability (also might want to do some error checking to make sure that the username is present)